draft-ietf-quic-transport.md

```diff
@@ -3374,6 +3367,36 @@ received on all streams, which are used to check for flow control violations. A
 receiver might use a sum of bytes consumed on all contributing streams to
 determine the maximum data limit to be advertised.
 
+
+### Stream 0 Flow Control
+
+The cryptographic handshake stream, Stream 0, is exempt from the
+connection-level data limits established by MAX_DATA.  That is, data sent on
+stream 0 is not counted against the limit expressed by MAX_DATA.  Data on stream
+0 is still subject to stream-level data limits and MAX_STREAM_DATA.
+
+Data sent in the Initial and Retry packets do not count toward stream 0 flow
+control limits.  These packet types do not permit the sending of data in
+multiple packets, so there is no opportunity for a peer to send MAX_STREAM_DATA
+frames.  Note that sending a Retry causes stream offsets and flow control to be
+reset for subsequent packets.  Thus, the only limit on the size of the
+cryptographic handshake messages these packets contain is determined by the MTU.
+
+Endpoints MUST respect the stream flow control limit set by their peer when
+sending STREAM frames for stream 0 in packets other than Initial or Retry.  A
+client might exceed stream flow control limits when sending an Initial packet
```

Comment:
Better.  A server can still fall into the same state, though.  Maybe just say "An endpoint might have reached or exceeded stream flow control limits when sending an Initial or Handshake packet and so...."